global_planner/src/planner_core.cpp

```diff
@@ -278,8 +278,17 @@ uint32_t GlobalPlanner::makePlan(const geometry_msgs::PoseStamped& start, const
         worldToMap(wx, wy, goal_x, goal_y);
     }
 
-    //clear the starting cell within the costmap because we know it can't be an obstacle
-    clearRobotCell(start, start_x_i, start_y_i);
+    // check if either start or goal poses are in collision
+    if (costmap_->getCost(start_x_i, start_y_i) >= costmap_2d::INSCRIBED_INFLATED_OBSTACLE) {
+      message = "The start pose is in collision. Planning from this pose will always fail";
+      ROS_ERROR_STREAM(message);
+      return mbf_msgs::GetPathResult::BLOCKED_START;
+    }
+    if (costmap_->getCost(goal_x_i, goal_y_i) >= costmap_2d::INSCRIBED_INFLATED_OBSTACLE) {
+      message = "The goal pose is in collision. Planning to this pose will always fail";
```

Comment:
F: this is only true if `default_tolerance` is set to `0`. I think we should instead return this if `found_legal` in `NavfnROS::makePlan` is false

Reply:
Q: We are using the `NavfnROS` planner, right?

Reply:
mmmm....  looks like default_tolerance is not used at all???
![image](https://user-images.githubusercontent.com/322610/209068543-a42b4a2f-867e-4117-81ae-6b6af8ea28ef.png)

Reply:
interesting... in that case the current change makes sense. Will take note in the [task about the default_tolerance](https://www.wrike.com/open.htm?id=1020458885)